### ISSUE TITLE
Structure Query Popover: broken location criteria

### ISSUE THREAD

**Issue by davilima6** [plone/mockup#678]
I failed trying to reproduce [this observation](https://github.com/plone/mockup/issues/381#issuecomment-43166686) from @djay, and actually noticed the behavior seems quite broken: https://dl.dropboxusercontent.com/u/5766075/screencasts/broken-location-criteria.mp4

Should we remove Location for 5.1 final while this isn't sorted out? It might take me a while to get familiar with this (Backbone) codebase.

**Comment by djay** [comment]
can’t seem to get your mp4 to play. can you run it through gifbrewery or similar so you can paste in in github? or write out the problem?

**Comment by davilima6** [comment]
Hi, @djay, thx for warning. The video's encoded in H264 so you need gstreamer1.0-libav in Ubuntu. I guess you're on OSX so I tested it and could watch the mp4 in VLC. Nevertheless here's the original file, this one should work ootb in browser:

**Comment by djay** [comment]
@davilima6 H264 shouldn't be an issue on osx but obviously some codec is having problems. You are right, VLC plays it fine. I've found it really useful to learn an animated gif too however. An issue is a record of the problem so it should be self-contained.
Gifbrewery doesn't seem to like webm either so I'd have to transcode it first :(

So looks like the issue is that the query function is ignoring relative path criteria (but not absolute paths). Do you know if the catalog is receiving the relative path criteria? Perhaps the context of the request is set to the root?

**Comment by davilima6** [comment]
Doesn't Gifbrewery open the mp4? Anyway VLC is really quick at transcoding, just Cmd+Shift+S.

And, no, I didn't have the time to debug it yet. Waiting for status:confirmed first ;-)

**Comment by djay** [comment]
yes, I worked it out. gif uploaded now (not sure what went wrong with the colours). I didn’t realise gif brewery is paid. I must have got it in the first 2 weeks by accident when it was free. There are plenty of others though, inc online converters.

**Comment by davilima6** [comment]
Gif brewery _really_ didn't like my mp4 so I found this site which did a better job: http://ezgif.com/video-to-gif

btw I also updated a video I uploaded yesterday to: https://github.com/plone/mockup/issues/671#issuecomment-222504072

**Comment by thet** [comment]
@davilima6 @djay the location query filter seems a bit useless where the filter is applied only on the current view.
:/

**Comment by jensens** [comment]
Closing as outdated. Mockup was completely rewritten for Plone 6 (jQuery/Backbone/RequireJS → ES6/Patternslib). Old structure query popover Backbone bug.
If you think this issue is still relevant in the new implementation, please reopen it. Thank you for the original report!
